Upgrade relm4 dependency in Cargo.toml: 0.10.0 -> 0.10.1. Update edition: 2021 -> 2024. Set rust-version to 1.92.

zekefast opened the pull request:
# Overview

Upgrade `relm4` dependency in `Cargo.toml`: `0.10.0` -> `0.10.1`. Update edition: `2021` -> `2024`. Set `rust-version` to `1.92`.

This changes allow to align relm4-template with latest release of "relm4" crate.

# Additional Changes
- Update `Cargo.lock`.
- Enable `gnome_49` feature for `relm4` dependency in `Cargo.toml`.

# Related Pull Requests
- Relm4/Relm4#844 (merge before current)

# TODO
- [ ] Update version of `relm4` to `0.10.2` once it is released to include Relm4/Relm4#844

zekefast committed:
Upgrade relm4 dependency in Cargo.toml: 0.10.0 -> 0.10.1. Update edition: 2021 -> 2024. Set rust-version to 1.92.

This changes allow to align relm4-template with latest release of "relm4" crate.

Additional changes:
- Update Cargo.lock.
- Enable "gnome_49" feature for relm4 dependency in Cargo.toml.

zekefast committed:
Format code.

bahrom04 commented on Cargo.toml:
```diff
@@ -2,15 +2,16 @@
 gettext-rs = { version = "0.7.7", features = ["gettext-system"] }
-relm4 = { version = "0.10.0", features = ["libadwaita", "gnome_48"] }
+relm4 = { version = "0.10.1", features = ["libadwaita", "gnome_49"] }
 adw = { version = "0.8.1", package = "libadwaita", features = ["v1_8"] }
```
```suggestion
```

this no longer helpful. gnome_49 provides compability with adw which previously we had to spesify version

zekefast replied:
Thank you for review and pointing this out!

There is a bug I introduced with addition of `gnome_49` flag. I enabled `libadwaita/v1_9` but Gnome 49 actually ships `libadwaita` 1.8. I added the fix #844.

Once it is merged and we release `0.10.2` I update this PR with appropriate fixes.

zekefast committed:
Remove dependency on `libadwaita` in Cargo.toml.

The dependency was added to specify correct version of `libadwaita` as `gnome_49` feature flag addition incorrectly enable `libadwaita/v1_9` while Gnome 49 ships with `libadwaita` 1.8.

As we made the fix for `gnome_49` flag in `relm4` we can remove the direct dependency on `libadwaita` with version specifications.

Co-authored-by: bahrom04 <116780481+bahrom04@users.noreply.github.com>

Cyncrovee commented:
Is it missing [this package?](https://packages.ubuntu.com/noble/libadwaita-1-dev)

bahrom04 commented:
> Is it missing [this package?](https://packages.ubuntu.com/noble/libadwaita-1-dev)

Yes. This 0.10.1 version requires not existing [v1_9 adw version](https://github.com/Relm4/Relm4/pull/844/changes#diff-92855fe08471597bacc341999e6b95c76a703eed0699c259e9c868c54351b51d). 

When 0.10.2 comes out it will be used latest v1_8 adw

bahrom04 commented:
Updated version: https://github.com/Relm4/relm4-template/pull/54
